src/brod_producer_buffer.erl

```diff
@@ -34,26 +36,24 @@
 
 %% keep data in fun() to avoid huge log dumps in case of crash etc.
 -type data() :: fun(() -> kafka_kv()).
--type call() :: brod_call_ref().
 
 -record(req,
-        { call :: call()
-        , data :: data()
+        { call_ref :: brod_call_ref()
+        , data     :: data()
+        , bytes    :: non_neg_integer()
         }).
 
--type send_fun() :: fun(([{binary(), binary()}]) -> {ok, corr_id()}).
--define(ERR_FUN, fun() -> erlang:error(bad_init) end).
-
 -record(buf,
-        { buffer_limit = 1        :: pos_integer()
-        , onwire_limit = 1        :: pos_integer()
-        , msgset_bytes = 1        :: pos_integer()
-        , send_fun     = ?ERR_FUN :: send_fun()
```

Comment:
i loved this send_fun...
could you please change it back ?
it makes tests way much easier.